ui/hooks/useTokensToSearch.js

```diff
@@ -66,7 +66,8 @@ export function getRenderableTokenData(
   const usedIconUrl =
     iconUrl ||
     (tokenList[tokenAddress] &&
-      `images/contract/${tokenList[tokenAddress].iconUrl}`);
+      `images/contract/${tokenList[tokenAddress].iconUrl}`) ||
```

Comment:
When is `iconUrl` expected to be missing? I thought it was present for all tokens, dynamic and static.

Confusingly, it seems to be missing from the static token type in the controller, but I can see it being assigned. But for dynamic tokens, it's declared in the type.

Reply:
So the token could also be [one that the user has imported manually](https://github.com/MetaMask/metamask-extension/blob/a8a1f7121f5cf35b6700e100cf7ae62cbe8016bc/ui/hooks/useTokensToSearch.js#L142) which is neither in our static json list nor in any of the aggregators. In this case it seems unlikely we would have an image for it but for tokens coming from this source it would be of [this type ](https://github.com/MetaMask/controllers/blob/26be9d21ebcc183eb418d4da48b8a685b43aa19f/src/assets/TokenRatesController.ts#L42) so an image property and no iconUrl property.

@Gudahtt

Reply:
Makes sense!